tesla_http_proxy/DOCS.md

```diff
@@ -14,38 +14,103 @@ servers: nginx_proxy/*.conf
+Using the Home Assistant iOS app, open the Addon Web UI and click **Enroll public key in your vehicle**. This should launch the Tesla app where it prompts for approval to "allow third-party access to your vehicle". If you have multiple vehicles, you'll need to do this on each of them.
+
 > [!NOTE]
 > Your Tesla app must already be key-paired with the car.
 
 Configure the [Tesla integration](https://github.com/alandtse/tesla) to use this proxy. It should pre-fill the Client ID, URL, and certificate for you by reading them from this addon.
 
+## Specific setup guides
+
+### Cloudflare
+
+Setting up Tesla HTTP Proxy through cloudflare involves the following steps (all the steps are made with the available addons in home assistant, nothing external. Also it avoids the necessity to open ports on the router):
+
+1. Install **Cloudflared** addon. Then configure like this and start the addon.
+
+```yml
+external_hostname: ha.example.com
+additional_hosts:
+  - hostname: tesla.example.com
+    service: https://homeassistant-internal.example.com
+```
+
+2. Check on the **Cloudflare dashboard** that the two CNAME DNS Records appeared. If not read the **cloudflared** logs for errors.
+   ![image](https://github.com/llamafilm/tesla-http-proxy-addon/assets/1372028/42e637a8-4428-41c7-b244-099aa6844216)
+3. Create an `A` type record with name `homeassistant-internal` and IPv4 address pointing to the `LOCAL IP ADDRESS` of the machine hosting the homeassistant instance. Disable `proxied`.
+4. Install **Let's Encrypt** addon. (can be done with other addons, check discussions on the repo for alternatives). Then configure like this and make sure to configure the `cloudflare_api_token`. Make sure the `keyfile` and `certfile` aren't already been created. If so, delete them and proceed starting the addon. Read the logs to see when the files are created (takes more than 60 seconds).
+
+```yml
+domains:
+  - ha.example.com
+  - "*.ha.example.com"
+  - tesla.example.com
+  - homeassistant-internal.example.com
+email: null@null.com
+keyfile: privkey.pem
+certfile: fullchain.pem
+challenge: dns
+dns:
+  provider: dns-cloudflare
+  cloudflare_api_token: redacted
+```
+
+4. Install **SSL Proxy** addon. (can be done with other addons, check discussions on the repo for alternatives). Then configure like this and make sure the paths to `certfile` and `keyfile` are correct. Start the addon. Make sure no errors appear in the logs.
+
+```yml
+domain: homeassistant-internal.example.com
+hsts: max-age=31536000; includeSubDomains
+certfile: fullchain.pem
+keyfile: privkey.pem
+cloudflare: true
+customize:
+  active: true
+  default: nginx_proxy_default*.conf
+  servers: nginx_proxy/*.conf
+```
+
+5. Install **Tesla HTTP Proxy** addon. Configure it as below and then follow all the steps in the `DOCS` making sure to fill the `client_id` and `client_secret`.
+
```

Comment:
Might be worth to mention the user to pick his `region`.

Reply:
I suggest removing this section, since most people will configure using the GUI, and instructions are already written above.